code/datums/flock/flock.dm

```diff
@@ -753,7 +766,7 @@ var/flock_signal_unleashed = FALSE
 					break
 			var/dir = O.dir
 			var/replacementPath = flock_conversion_paths[keyPath]
-			var/obj/converted = new replacementPath(T)
+			var/obj/converted = replacementPath != /obj/flock_structure/fabricator ? new replacementPath(T) : new replacementPath(T, null, O)
```

Comment:
I still think it would make more sense to just pass this to all flock structures as there are likely to be more structures added that need to change based on the thing they were converted from. Switching arguments around and giving them different meanings in each subtype is OOP crime.

Reply:
istype() wasn't working here, had to check it like this, wasn't aware of any other way to fix it than do it outside of New()

Reply:
Just don't do the check and pass it to all flock structures. `istype` would also be crime.